doc/mgmt/gnmi/gnsi.md

```diff
@@ -0,0 +1,383 @@
+- Is this change specific to any platform? Are there dependencies for platforms to implement anything to make this feature work? If yes, explain in detail and inform community in advance.
+  - Any Platform
+- SAI API requirements, CLI requirements, ConfigDB requirements. Design is covered in following sections.
+
+#### Certz
+
+##### Certz.Rotate
+
+The Certz service definition provides the API to be used for rotating and testing PKI primitives used on network systems. The PKI primitives are grouped by SSL profiles and the service can manage multiple profiles beyond the default.
+
+The Rotate() function is a bidirectional streaming RPC which permits mutating Certificates, Root Certificate Bundles, Certificate Revocation Lists and Authentication Policies of a given profile. In a Rotate() stream it is possible to mutate one or more of the elements, and to send a Finalize message once the in-flight change has been verified to be operational. Failure to send the Finalize message will result in the candidate element being discarded and the original element being used instead.
+
+Certz.Rotate() will permit rotation, and verification of function, of any of the PKI elements. The normal use-case would be to:
+- send a PKI element to a network system as a RotateCertificateRequest.
+- verify that the services which will use the new certificate bundle continue to operate normally.
+- send a FinalizeRequest to finish the rotation process.
+
+The Certz Rotate Flow Diagram
+![Rotate Flow Chart](images/gnsi_certz_rotate.png)
+
+##### Profiles
+
+SSL profiles logically group the PKI primitives together and any gRPC Server may be configured to use a given profile. There is a default profile with the `gnxi` id that gNMI/gNOI/gNSI use. If no profile is specified, this is the profile that will be referenced.
+
+There are 3 RPC for profiles:
+- Certz.GetProfileList() - Existing profiles on a target can be discovered.
+- Certz.AddProfile() - An SSL profile can be added.
+- Certz.DeleteProfile() - When a profile is no longer needed it can be removed. Note that the gnxi SSL profile cannot be removed.
+
+##### Certificate Signing Request
+
+The server can optionally support generating a Certificate Signing Request. This allows the client to open a Certz.Rotate() stream, request a CSR and then have the Certificate signed externally.
+
+- Certz.CanGenerateCSR()
+- Certz.Rotate(GenerateCSRRequest)
+
+#### Authz
+Authz is policy based authorization for gRPC access. The policy is enforced by the gRPC server by registering a file watcher with a gRPC Interceptor in the options. The policy is managed by the gNSI Authz service running in the gNMI server. \
+
+The policy to be enforced is defined in the form of a JSON string whose schema can be found [here](https://github.com/grpc/proposal/blob/master/A43-grpc-authorization-api.md).
+
+- Authz.Rotate()
+  - upload a new policy - same flow as Certz diagram
+- Authz.Probe()
+  - test the current policy
```

Comment:
Nits: can we be a bit more specific here? For example, are we simply test for existence or something more complex? Similar for Pathz.Probe.